src/useCookieConsent.ts

```diff
@@ -2,49 +2,44 @@ import { useState, useEffect } from 'react';
 import * as Cookies from 'js-cookie';
 import { COOKIE_CONSENT_KEY, DEFAULT_CONSENT } from './constants';
 import {
-  CookieConsent,
-  ConsentHookStateType,
-  CookieConsentOptions,
+  UseCookieConsentType,
+  ConsentState,
   CookieWrapper,
   DidDeclineAllHandler,
   AcceptCookiesOptions,
 } from './types';
 import { allCookiesSetToValue, allPropsApproved } from './utils';
 
-export const useCookieConsent = (
-  options?: CookieConsentOptions
-): ConsentHookStateType => {
+export const useCookieConsent: UseCookieConsentType = (options) => {
   const doesStorageExist =
     typeof options !== 'undefined' && 'storage' in options;
 
   const storedConsent = doesStorageExist
     ? JSON.parse(options?.storage.getItem(COOKIE_CONSENT_KEY))
     : Cookies.getJSON(COOKIE_CONSENT_KEY);
 
-  const initialConsent: CookieConsent =
+  const initialConsent: ConsentState =
     storedConsent || options?.defaultConsent || DEFAULT_CONSENT;
 
-  const [consent, setConsent] = useState<CookieConsent>(initialConsent);
+  const [consent, setConsent] = useState<ConsentState>(initialConsent);
 
   useEffect(() => {
-    if (consent?.necessary) {
-      if (doesStorageExist) {
-        options.storage.setItem(COOKIE_CONSENT_KEY, JSON.stringify(consent));
-      } else if (
-        typeof options === 'undefined' ||
-        'consentCookieAttributes' in options
-      ) {
-        Cookies.set(
-          COOKIE_CONSENT_KEY,
-          consent,
-          options?.consentCookieAttributes
-        );
-      }
+    if (consent?.necessary && doesStorageExist) {
```

Comment:
For the `consent?.necessary`, the idea was **to not run anything** if it's falsy. Now you're actually running the code inside of `else if`. You instead need to early-escape it. Just inverse the logic statement instead and return early: `if (!consent?.necessary) return;`